hera_cal/lstbin.py

```diff
@@ -579,110 +577,137 @@ def lst_bin_files(data_files, filetype='uvh5', input_cals=None, dlst=None, verbo
+                            utils.echo("Opening and applying {}".format(input_cals[j][k]), verbose=verbose)
+                            uvc = io.to_HERACal(input_cals[j][k])
                             gains, cal_flags, quals, totquals = uvc.read()
-                            apply_cal.calibrate_in_place(d, gains, data_flags=fl, cal_flags=cal_flags,
+                            # down select times in necessary
+                            if False in tinds and uvc.Ntimes > 1:
+                                # If uvc has Ntimes == 1, then broadcast across time will work automatically
+                                uvc.select(times=uvc.time_array[tinds])
+                                gains, cal_flags, quals, totquals = uvc.read()
+                            apply_cal.calibrate_in_place(data, gains, data_flags=flags, cal_flags=cal_flags,
                                                          gain_convention=uvc.gain_convention)
 
-                    # unwrap li relative to itself
-                    li[li < li[0]] += 2 * np.pi
-
-                    # unwrap li relative to start_lst
-                    li[li < start_lst - atol] += 2 * np.pi
+                    file_list.append(data_files[j][k])
+                    nightly_data_list.append(data)  # this is data
+                    nightly_flgs_list.append(flags)  # this is flgs
+                    nightly_lst_list.append(larr[tinds])  # this is lsts
 
-                    # pass data references to data_status list
-                    data_status[j][k] = [d, fl, ap, a, f, t, li, p]
+                # skip if nothing accumulated in nightly files
+                if len(nightly_data_list) == 0:
+                    continue
 
-                    # erase references from namespace
-                    del d, fl, ap, a, f, t, li, p
+                # extend to data lists
+                data_list.extend(nightly_data_list)
+                flgs_list.extend(nightly_flgs_list)
+                lst_list.extend(nightly_lst_list)
 
-                elif not f_select[j][k] and old_f_select[j][k]:
-                    # close file
-                    del data_status[j][k]
-                    data_status[j].insert(k, None)
+                del nightly_data_list, nightly_flgs_list, nightly_lst_list
 
-                # if file is needed, append data references to data_list
-                if f_select[j][k]:
-                    file_list.append(data_files[j][k])
-                    nightly_data_list.append(data_status[j][k][0])  # this is data
-                    nightly_flgs_list.append(data_status[j][k][1])  # this is flgs
-                    nightly_lst_list.append(data_status[j][k][6])  # this is lsts
-
-            # skip if nothing accumulated in nightly files
-            if len(nightly_data_list) == 0:
+            # skip if data_list is empty
+            if len(data_list) == 0:
+                # erase data references
+                del data_list, flgs_list, lst_list
```

Comment:
I believe this is unnecessary. These get overwritten with [] and therefore should get garbage collected.